Feat dashboard

ryanntannn opened the pull request:
## Describe your changes

- styling overhaul for landing page, dashboard, elder CRUD and caregiver CRUD

## Steps to test

- npm run clean
- npm run migrate
- test onboarding flow
- create a new elder
- try invite flow (might require separate singpass account)
- edit elder details
- edit caregiver details


## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] I have added necessary documentation

ryanntannn commented on packages/frontend/src/components/ui/address-form.tsx:
```diff
@@ -0,0 +1,305 @@
+    }
+
+    const script = document.createElement("script");
+    script.src = `https://maps.googleapis.com/maps/api/js?key=${env.GOOGLE_MAPS_API_KEY}&libraries=places`;
```
ignore

Kantosaurus committed:
Implemented focus-bento

Kantosaurus committed:
Refined new caregiver page

Kantosaurus committed:
chore: update package-lock.json dependencies

Kantosaurus committed:
feat: update frontend landing page and dependencies

- Modified frontend landing page component
- Updated frontend package.json dependencies
- Updated package-lock.json with new dependency versions

Kantosaurus committed:
upgraded framer-motion to motion/react

Kantosaurus committed:
feat: enhance elder profile and form functionality

- Update elder form component with improved validation and UI
- Enhance elder profile page with better user experience
- Add invite link generation and sharing capabilities
- Implement map integration for location display
- Improve overall elder management interface

ryanntannn committed:
fix: many issues

Kantosaurus committed:
feat: add elder profile and edit functionality with DOB support

- Add elder profile page with detailed view and edit capabilities
- Implement elder edit page with form validation
- Add caregiver profile page
- Add DOB field to elder entity and database schema
- Update elder forms to include DOB validation
- Enhance dashboard with elder cards and navigation
- Add new elder profile and edit routes
- Include elder profile image placeholder
- Update database migrations for elder DOB field

ryanntannn committed:
style: ui

shansjx reviewed (changes requested):
caregiver profile page does not seem to immediately display updated caregiver information once edit is saved and navigated back
Need to do a manual refresh to see the changes. Would be good for immediate refresh so it is displayed immediate changes to the user. Other than than lgtm

Kantosaurus committed:
fixed the error: duplicate key value violates unique constraint "caregivers_pkey"
    at Parser.parseErrorMessage (C:\Users\wooai\Documents\ESC-C3T2\node_modules\pg-protocol\src\parser.ts:369:69)
    at Parser.handlePacket (C:\Users\wooai\Documents\ESC-C3T2\node_modules\pg-protocol\src\parser.ts:187:21)
    at Parser.parse (C:\Users\wooai\Documents\ESC-C3T2\node_modules\pg-protocol\src\parser.ts:102:30)
    at Socket.<anonymous> (C:\Users\wooai\Documents\ESC-C3T2\node_modules\pg-protocol\src\index.ts:7:48)
    at Socket.emit (node:events:507:28)
    at addChunk (node:internal/streams/readable:559:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
    at Readable.push (node:internal/streams/readable:390:5)
    at TCP.onStreamRead (node:internal/stream_base_commons:189:23)

added to the caregiver entity to check if the caregiver is already in the database

Kantosaurus committed:
fix elder schema and bento grid focus
